plugins/debuginfo-install.py

```diff
@@ -79,54 +83,95 @@ def run(self):
             raise dnf.exceptions.PackagesNotAvailableError(_("Unable to find a match"),
                                                            pkg_spec=' '.join(errors_spec))
 
-    def _dbg_available(self, dbgname, package, match_evra):
-        if match_evra:
-            return self.packages_available.filter(
-                name=dbgname,
-                epoch=int(package.epoch),
-                version=str(package.version),
-                release=str(package.release),
-                arch=str(package.arch))
-        else:
-            return self.packages_available.filter(
-                name=dbgname,
-                arch=str(package.arch))
-
-    def _di_install(self, package):
-        for dbgname in [package.debug_name, package.source_debug_name]:
-            if dbgname in self.dbgdone:
-                break
-            if self._dbg_available(dbgname, package, True):
-                di = "{0}-{1}:{2}-{3}.{4}".format(
-                    dbgname,
-                    package.epoch,
-                    package.version,
-                    package.release,
-                    package.arch)
-                self.base.install(di)
-            elif self._dbg_available(dbgname, package, False):
-                di = "{0}.{1}".format(dbgname, package.arch)
-                self.base.install(di)
-            else:
-                continue
-            self.dbgdone.append(dbgname)
-            break
-        else:
-            logger.info(
-                _("Could not find debuginfo for package: %s") % package)
+    def _install(self, package, debug_name, base_nevra):
+        if debug_name in self.packages_installed:
+            return True
+
+        kwargs = {}
+
+        # if some part of EVRA was specified in the argument, add it as a filter
+        if base_nevra.epoch is not None:
+            kwargs["epoch"] = base_nevra.epoch
+        if base_nevra.version is not None:
+            kwargs["version"] = base_nevra.version
+        if base_nevra.release is not None:
+            kwargs["release"] = base_nevra.release
+        if base_nevra.arch is not None:
+            kwargs["arch"] = base_nevra.arch
+
+        # if the package is installed and none of EVR was specified in the
+        # argument, use the installed package's EVR (install debuginfo with the
+        # same version as the installed package)
+        if package._from_system and \
+                ("epoch" not in kwargs or kwargs["epoch"] == package.epoch) and \
+                ("version" not in kwargs or kwargs["version"] == package.version) and \
+                ("release" not in kwargs or kwargs["release"] == package.release):
+            kwargs["epoch"] = package.epoch
+            kwargs["version"] = package.version
+            kwargs["release"] = package.release
+
+        query = self.packages_available.filter(name=debug_name, **kwargs)
+
+        if query:
+            s = dnf.selector.Selector(self.base.sack)
```

Comment:
Please could you rename s to sltr or something else? It will look better.

Reply:
For this three-line scope and clear definition of what `s` is on the right side of `=` I think one letter variables are fine, but I've renamed it to `selector`.